### DISCUSSION TITLE
Unique element ids

### DISCUSSION THREAD

**Discussion by djkunkel** [vueform/vueform#86] [Questions] [answered]
Is there a way to have VueForm prefix all element ids with something so they don't conflict with already existing element ids on a page? I didn't see any documentation or examples around this.

**Comment by adamberecz** [comment, marked as answer]
You can create a [plugin](https://vueform.com/docs/plugins#plugins) that extends the [`fieldId`](https://github.com/vueform/vueform/blob/main/src/composables/elements/useFieldId.js#L21) for all elements:

```js
// vueform.config.js

import { computed } from 'vue'

export default {
  // ...
  plugins: [
    {
      apply: /^[a-zA-Z]+Element$/,
      setup(props, context, component) {
        const fieldId = computed(() => {
          return `prefix-${component.fieldId.value}`
        })

        return {
          ...component,
          fieldId,
        }
      }
    }
  ]
}
```

You might want to outsource the plugin itself to a separate file to eg. `./vueform/plugins/id-prefix-plugin.js`.

**Reply by djkunkel** [reply]
This will work well for me. Thank you. I did add a check to make sure there was no parent so it wouldn't double prefix grouped elements.
```
if(!component.parent.value) {
    return `prefix-${component.fieldId.value}`
} else {
    return component.fieldId.value
}
```
